examples/browser-webrtc/server/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+//! Builds the server, exported so that bin/main.rs can run it
+use anyhow::Result;
+use axum::{http::Method, routing::get, Router};
+use futures::StreamExt;
+use libp2p_core::muxing::StreamMuxerBox;
+use libp2p_core::Transport;
+use libp2p_identity as identity;
+use libp2p_ping as ping;
+use libp2p_relay as relay;
+use libp2p_swarm::{keep_alive, NetworkBehaviour, SwarmBuilder, SwarmEvent};
+use libp2p_webrtc as webrtc;
+use multiaddr::{Multiaddr, Protocol};
+use rand::thread_rng;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr};
+use tower_http::cors::{Any, CorsLayer};
+use void::Void;
+
+pub const PORT: u16 = 4455;
+
+pub async fn start(remote: Option<Multiaddr>) -> Result<()> {
+    let id_keys = identity::Keypair::generate_ed25519();
+    let local_peer_id = id_keys.public().to_peer_id();
+    let transport = webrtc::tokio::Transport::new(
+        id_keys,
+        webrtc::tokio::Certificate::generate(&mut thread_rng())?,
+    )
+    .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn)))
+    .boxed();
+
+    let behaviour = Behaviour {
+        relay: relay::Behaviour::new(local_peer_id, Default::default()),
+        ping: ping::Behaviour::new(ping::Config::new()),
+        keep_alive: keep_alive::Behaviour,
+    };
+
+    let mut swarm = SwarmBuilder::with_tokio_executor(transport, behaviour, local_peer_id).build();
+
+    let address_webrtc = Multiaddr::from(Ipv6Addr::LOCALHOST)
+        .with(Protocol::Udp(0))
+        .with(Protocol::WebRTCDirect);
+
+    swarm.listen_on(address_webrtc.clone())?;
+
+    // Dial the peer identified by the multi-address given as the second
+    // command-line argument, if any.
+    if let Some(addr) = remote {
+        log::info!("Dialing {addr}");
+        swarm.dial(addr)?;
+    }
```

Comment:
When does the server ever need to dial another server?

Reply:
It never does in this example, I'm remvoing it